test_rearrange_cars.py

```diff
@@ -0,0 +1,72 @@
+import logging
+import unittest
+import numpy as np
+from rearrange_cars_luiza import RearrangeCars
+
+class TestRearrangeCars(unittest.TestCase):
+
+    '''Unittests for RearrangeCars class and promised fuctionality
+    '''
+
+    def setUp(self):
+
+        '''Basic setting that will be used in several tests
+        '''
+        self.init_order = [1, 2, 0, 3]
+        self.final_order = [3, 1, 2, 0]
+        self.rc = RearrangeCars(self.init_order, self.final_order)
+
+    def test_rearrange_all_cars(self):
+
+        '''Tests on the toy example without special difficulties
+        '''
+
+        self.rc.rearrange_all_cars()
+        self.assertEqual(self.rc.current_car_order, self.final_order)
+
+    def test_rearrange_all_cars_twice(self):
+
+        '''Test whether the class instance can properly handle
+        running the same function twice
+        '''
+
+        self.rc.rearrange_all_cars()
+        self.assertEqual(self.rc.current_car_order, self.final_order)
+        self.rc.rearrange_all_cars()
+        self.assertEqual(self.rc.current_car_order, self.final_order)
+
+    def test_already_proper_order(self):
+
+        '''Tests the case when initial and final orders are the same
+        '''
+
+        rc_same = RearrangeCars(self.final_order, self.final_order)
+        self.assertRaises(rc_same.rearrange_all_cars())
```

Comment:
I was wondering another thing... Does it matter that the initial arrangement is the same as the desired one? Perhaps then we can just return an empty list (if the rearrange function is to return list of course XD)?